### PR TITLE
fix: proper empty admin key validation

### PR DESCRIPTION
**Pull request by fekitibi**
### Description

APISIX now prevents startup when admin key validation is enabled (admin_key_required: true) but no valid admin keys are configured. This enhancement improves security by ensuring proper authentication is always available for admin API access.

This PR is still work in progress, the tests are failing because of the new approach(many tests had set admin keys to ~ or null which dont let apisix to start up properly.

#### Which issue(s) this PR fixes:

Fixes #12484 

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by fekitibi** [comment]
@SkyeYoung 

The tests are failing after my changes. Here’s why: when admin keys are enabled, adding an invalid admin key now causes APISIX to exit. To handle this, I updated APISIX.pm (line 972) so that admin keys are disabled by default in tests.

This works for tests without a deployment section, but many tests that do have a deployment section still enable admin keys. In those cases, the keys are often defined as "~" or "null", which breaks the tests.

What would be the best way to approach this? Any tips would be greatly appreciated.

**Comment by fekitibi** [comment]
I’ve tested this from my own fork, and all tests are passing 😄

**Comment by fekitibi** [comment]
This is so weird, I have ran the same CI workflows on my fork, and they passed without any errors: https://github.com/fekitibi/apisix/actions/runs/17134962910

Will try to rerun them on my fork and then I will get back here.

**Comment by fekitibi** [comment]
@SkyeYoung 

I fixed all the tests that failed:
Code Lint passed: https://github.com/fekitibi/apisix/actions/runs/17179691316
CI passed(but I had to run it 4 times to pass every tests): https://github.com/fekitibi/apisix/actions/runs/17155412073
CLI test passed: https://github.com/fekitibi/apisix/actions/runs/17155412082
Source install passed: https://github.com/fekitibi/apisix/actions/runs/17155412097

I have modified the CLI tests added an admin key since we are not autogenerating them, I was wondering if there is a better way to add them in one place instead of midifying all of the effected CLI tests.

For the source install changes I am not sure if they are okey it seems like they were missing some permissions, then I had to fix the syntax. If these are innecessary changes to the source-install workflow I can try to revert them.

**Comment by juzhiyuan** [comment]
[Just approved CI to run]

**Comment by fekitibi** [comment]
@juzhiyuan  

It looks like 3 tests failed. Could you please re-run the CI tests?  
On my fork, I had to re-run them a few times (about 4) before they passed. Not sure why this happens.

**Comment by Baoyuantop** [comment]
There is a CI failed but it looks like a known issue and not related to this PR.

**Review by Baoyuantop — COMMENTED** [review]
Hi @fekitibi, could you please correct your pull request description? I don't see the `admin_key_required` configuration changes and `APISIX enters a restart loop` part in the code changes. The pull request description should accurately reflect the technical solution you implemented, which will also make it easier for other maintainers to review.

**Comment by fekitibi** [comment]
I will update the description later today 😄

**Comment by fekitibi** [comment]
@Baoyuantop  I just updated the description of the PR, I hope it is better now 😄 .

**Review by Baoyuantop — COMMENTED** [review]
1. Some scenarios need to be added to the test, such as testing the skip logic of empty keys, testing mixed configurations (valid keys and empty keys coexist), etc.

2. The auto-generated logic has been completely removed. I don't know if this is considered a breaking change? cc @bzp2010

**Comment by fekitibi** [comment]
@Baoyuantop I just added the new test cases, I hope this will be sufficient 😄 
Let me know if I should change anything.

File: conf/config.yaml [hunk 1]
```diff
@@ -59,5 +59,5 @@ deployment:
   admin:
     admin_key:
       - name: admin
-        key: ''  # using fixed API token has security risk, please update it when you deploy to production environment. If passed empty then will be autogenerated by APISIX and will be written back here. Recommended is to use external mechanism to generate and store the token.
+        key: admin  # using fixed API token has security risk, please update it when you deploy to production environment. If passed empty then will be autogenerated by APISIX and will be written back here. Recommended is to use external mechanism to generate and store the token.
```
Comment (Baoyuantop) [inline conf/config.yaml:62]: This description needs to be changed.
